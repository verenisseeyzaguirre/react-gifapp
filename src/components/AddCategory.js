import React from 'react'
import { useState } from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ( { setCategories } ) => {

    const [inputValue, setInputValue] = useState(''); // si no habia '' era undefined y salia error

    const handleInputChange = ( e ) => {
        setInputValue( e.target.value )
        // console.log(e.target.value)
        console.log('handleInputChange llamado');
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        console.log('handleSubmit llamado', inputValue)

        // cats es el estado anterior
        if ( inputValue.trim().length > 2 ) {
            setCategories( cats => [inputValue, ...cats] );
            setInputValue('');
        }
        
    }

    return (
        <form onSubmit= {handleSubmit}>
            <input 
                type="text"
                value = { inputValue }
                onChange = { handleInputChange }
                placeholder= 'Buscar gif...'
            />
        </form>
    )

}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
  };