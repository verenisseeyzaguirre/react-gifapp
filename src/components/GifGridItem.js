import React from 'react'


import PropTypes from 'prop-types';
//https://animate.style/

// param ( img ) en p {img.title}
export const GifGridItem = ( { title, url } ) => {

    //console.log( img );

    return (
        <div className='card animate__animated animate__fadeIn'>
            <img src={ url } alt={ title }/>
            <div className='name'>
                <p>{ title }</p>
            </div>
        </div>
    )
}

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
};

