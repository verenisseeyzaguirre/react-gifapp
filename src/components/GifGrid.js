import React from 'react'
//, { useState, useEffect }
import { GifGridItem } from './GifGridItem';
//import { getGifs } from '../helpers/getGifs';
import { useFetchGifs } from '../hooks/useFetchGifs';

export const GifGrid = ( { category } ) => {

    //const [count, setCount] = useState(0);
    //const [images, setImages] = useState([]);

    const { data:images, loading } = useFetchGifs( category );

    // [] indica q se ejecuando cuando se renderiza por primera vez y ya no mas
    // salio un warning porque si cambia la categoria deberia renderizar nuevmnte
    /*
    useEffect( () => {
        getGifs( category )
            .then( setImages );
    }, [ category ])
    */
    // getGifs()

    return (
        <>
            <h3 className='animate__animated animate__fadeIn' > {category} </h3>

            { loading && <p className='animate__animated animate__flash' >Loading</p> }

            <div className='card-grid' >
                {
                    images.map( img => (
                        <GifGridItem 
                            key={img.id}
                            {...img}
                        />
                    ))
                }         
            </div>
        </>
    )
}

/*
    debajo de h3
    <h2>{count}</h2>
    <button onClick= {()=> setCount(count+1) }>+1</button>
*/

/*
<ol>
    {
        images.map( img => (
            <li  key={img.id}>{img.title}</li>
        ))
    }
</ol>
*/

/*
{
    images.map( img => (
        <GifGridItem 
            key={img.id}
            {...img}
        />
    ))
}  
*/