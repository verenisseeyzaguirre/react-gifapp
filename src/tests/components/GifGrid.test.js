import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';

import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
//fingir la llamada a ese archivo y suponer o controlar la info q eso va a responder
jest.mock( '../../hooks/useFetchGifs' );

describe('Pruebas en <GifGrid />', () => {

    const category = 'Bulma';
    

    test('Debe mostrar <GifGrid /> correctamente', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow( <GifGrid  category = {category}/> )
        expect( wrapper ).toMatchSnapshot();

    });

    test('Debe mostrar items cuando se cargan imagenes useFetchGifs', () => {

        const gifs = [{
            id: '123',
            title: 'Titulo',
            url: 'https://localho.com'
        },
        {
            id: '124',
            title: 'Titulo',
            url: 'https://localho.com'
        }];
        
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });

        const wrapper = shallow( <GifGrid  category = {category}/> )
        //expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('p').exists() ).toBe(false);
        expect( wrapper.find('GifGridItem').length ).toBe(gifs.length);
        
    });

})

/*
let wrapper;
    let gifs = [];
 
    beforeEach(() => {
        jest.clearAllMocks();
        gifs = [{
            id: 'ABC',
            url: 'https://localhost/cualquier/cosa.jpg',
            title: 'Cualquier Cosa'
        },
        {
            id: '123',
            url: 'https://localhost/cualquier/cosa.jpg',
            title: 'Cualquier Cosa'
        }]
 
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false,
        });
        wrapper = shallow(<GifGrid category={category} />);
    })
*/