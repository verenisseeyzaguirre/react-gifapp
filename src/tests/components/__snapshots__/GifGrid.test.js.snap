// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid /> Debe mostrar <GifGrid /> correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    Bulma
     
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
